pkg/templates/aws/resources/s3_bucket.yaml

```diff
@@ -2,22 +2,17 @@ qualified_type_name: aws:s3_bucket
 display_name: S3 Bucket
 sanitize_name:
   # https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
-  # Identifiers have these naming constraints:
-  # - Must contain 1–63 alphanumeric characters or hyphens (55 since we reserve 8 for an IaC-appended suffix).
-  # - First character must be a letter.
-  # - Can't end with a hyphen or contain two consecutive hyphens.
-  # - Names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).
   |
   {{ . 
-    | replace `^[^[:alpha:]]+` "" 
-    | replace `--+` "-" 
-    | replace `-$` ""
-    | replace `_` "-"
-    | replace `[^[:alnum:]-]+` "-"
-    | length 1 55
+    | replace `^(xn--)|sthree-)|(sthree-configurator)` ""
+    | replace `(-s3alias)|(--ol-s3)$` ""
+    | replace `[^a-z0-9.-]+` "-"
+    | replace `(^[.-]+)|([.-]+$)` ""
+    | replace `\.\.+` "."
+    | replace `^([0-9]+)\.([0-9]+)\.([0-9]+)` "$1-$2-$3"
+    | length 3 63
```

Comment:
probably want length 55 for the iac suffix

Reply:
Isn't that a pulumi-specific implementation detail?